system/src/system_task.cpp

```diff
@@ -285,6 +286,16 @@ void establish_cloud_connection()
             return;
         }
 
+#if PLATFORM_ID==PLATFORM_ELECTRON_PRODUCTION
```

Comment:
rather than hard-coding the provider port, APN and other details. it would be more flexible and less code to create structs in the HAL that describe these things as data, keyed on the provider ID, so that we don't have any conditional logic for each provider.